tr55/model.py

```diff
@@ -126,62 +127,27 @@ def clamp(runoff, et, inf, precip):
     precip = max(0.0, precip)
     soil_type, land_use, bmp = cell.lower().split(':')
 
-    # If there is no precipitation, then there is no runoff or
-    # infiltration.  There is evapotranspiration, however (it is
-    # understood that over a period of time, this can lead to the sum
-    # of the three values exceeding the total precipitation).
+    # If  there  is no  precipitation,  then  there  is no  runoff  or
+    # infiltration;  however,  there  is evapotranspiration.   (It  is
+    # understood that over a period of  time, this can lead to the sum
+    # of the three values exceeding the total precipitation.)
     if precip == 0.0:
         return {
             'runoff-vol': 0.0,
-            # 'et-vol': cell_count * evaptrans,
             'et-vol': 0.0,
             'inf-vol': 0.0,
         }
 
-    # Deal with the Best Management Practices (BMPs).  For most BMPs,
-    # the infiltration is read from the table and the runoff is what
-    # is left over after infiltration and evapotranspiration.  Rain
-    # gardens are treated differently.
-    if bmp and is_bmp(bmp) and bmp != 'rain_garden':
-        inf = lookup_bmp_infiltration(soil_type, bmp)  # infiltration
-        runoff = max(0.0, precip - (evaptrans + inf))  # runoff
-        (runoff, evaptrans, inf) = clamp(runoff, evaptrans, inf, precip)
-        return {
-            'runoff-vol': cell_count * runoff,
-            'et-vol': cell_count * evaptrans,
-            'inf-vol': cell_count * inf
-        }
-    elif bmp and bmp == 'rain_garden':
-        # Here, return a mixture of 20% ideal rain garden and 80%
-        # high-intensity residential.
-        inf = lookup_bmp_infiltration(soil_type, bmp)
-        runoff = max(0.0, precip - (evaptrans + inf))
-        hi_res_cell = soil_type + ':developed_med:'
-        hi_res = simulate_cell_day(precip, evaptrans, hi_res_cell, 1)
-        hir_run = hi_res['runoff-vol']
-        hir_et = hi_res['et-vol']
-        hir_inf = hi_res['inf-vol']
-        final_runoff = (0.2 * runoff + 0.8 * hir_run)
-        final_et = (0.2 * evaptrans + 0.8 * hir_et)
-        final_inf = (0.2 * inf + 0.8 * hir_inf)
-        final = clamp(final_runoff, final_et, final_inf, precip)
-        (final_runoff, final_et, final_inf) = final
-        return {
-            'runoff-vol': cell_count * final_runoff,
-            'et-vol': cell_count * final_et,
-            'inf-vol': cell_count * final_inf
-        }
-
-    # At this point, if the `bmp` string has non-zero length, it is
-    # equal to either 'no_till' or 'cluster_housing'.
-    if bmp and bmp != 'no_till' and bmp != 'cluster_housing':
-        raise KeyError('Unexpected BMP: %s' % bmp)
-    land_use = bmp or land_use
+    # If  the BMP  is cluster_housing  or  no_till, then  make it  the
+    # land-use.  This is  done because those two types  of BMPs behave
+    # more like land-uses than they do BMPs.
```

Comment:
I just recently re-discovered this feature in emacs.  I think it used to either have a different keybinding or a different behavior vis-a-vis regions (or something) back in emacs 22, and I guess I thought the feature had been removed when I couldn't activate it.  Now that I have rediscovered it, I am using it with gusto.